code/modules/shuttle/supply.dm

```diff
@@ -690,6 +690,34 @@
 			if(i == 1)
 				O.generateRequisition(loc)
 
+	else if (href_list["rejectall"])
```

Comment:
This section, from this line 693 down to 719 is overly complicated, when it can actually be a lot simpler. I pulled your code and made some tweaks:
```js
	else if(href_list["rejectall"])
		for(var/A in SSshuttle.requestlist)
			var/datum/supply_order/SO = A
			notify_pda(SO.orderedby, SO.ordernum, SO.object.name)
		SSshuttle.requestlist.Cut()
```
Everything still worked fine, all the orders got rejected, they disappeared from the cargo console list, and I got a bunch of rejected PDA notifications. You might need to tweak your PDA proc a bit to make this work but its much easier to read.

Reply:
This is how i did it initially, however the downside of the simple method is that when you request a lot of crates you get bombarded by a lot of notifications and beeps in a split second. It floods the chat screen with messages and the beep sounds overlap and sound not very cool.
So what i tried to accomplish is to group requests from the same person of the same item into one message. But unfortunally the code turns out to be complicated.

Reply:
Oh jeez, ok then. But I have to wonder if putting up with a bit of spam is fine just for the sake of code simplicity though, because this probably won't happen more than 1 or 2 times a round, if even that. 

I suppose you could also find a way to just send a no-reply PDA message they can view, so its just one message and one ping, but that has a list of all the rejected orders. Just some ideas though, you don't have to listen to me.

Reply:
I will definitely add some comments in the code for now to make it easier to read. Will see what other reviewers say.

Reply:
Might be best to make this into a proc of it's own seeing the size of the logic done in here.